### PR TITLE
Ray-circle intersects function returns both intersection points now

### PR DESCRIPTION
**Pull request by daic0r**

### CODE REVIEW

File: olcUTIL_Geometry2D.h [hunk 1]
```diff
@@ -2457,7 +2457,8 @@ namespace olc::utils::geom2d
 			if (s2 < 0)
 				return { q.origin + q.direction * s1 };
 
-			return { q.origin + q.direction * std::min(s1, s2) };
+			const auto minmax = std::minmax(s1, s2);
```
Comment (cugone) [inline olcUTIL_Geometry2D.h:2460]: I would recommend using structured bindings here to facilitate a better name than just the function name itself to improve readability:

```cpp
const auto& [first_intersection_distance, second_intersection_distance] = std::minmax(s1, s2);
return {q.origin + q.direction * first_intersection_distance, q.origin + q.direction * second_intersection_distance};
//...
```

Comment (daic0r) [reply olcUTIL_Geometry2D.h:2460]: Good point! At my day job I am responsible for the PR reviews and I would've commented the exact same thing, so I don't know why I didn't do it in the first place :D

Comment (daic0r) [reply olcUTIL_Geometry2D.h]: Done. Left out the reference, though, since we're dealing with fundamental types.

**Comment by OneLoneCoder** [comment]
Yeah like this, thanks!

**Pull request merged**